test/Npgsql.Tests/Types/TsQueryTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections;
+using System.Threading.Tasks;
+using NpgsqlTypes;
+using NUnit.Framework;
+
+namespace Npgsql.Tests.Types
+{
+    [TestFixture(MultiplexingMode.NonMultiplexing, false)]
+    [TestFixture(MultiplexingMode.NonMultiplexing, true)]
+    [TestFixture(MultiplexingMode.Multiplexing, false)]
+    [TestFixture(MultiplexingMode.Multiplexing, true)]
+    public sealed class TsQueryTests : TypeHandlerTestBase<NpgsqlTsQuery>
+    {
+        public TsQueryTests(MultiplexingMode multiplexingMode, bool useTypeName) : base(
+            multiplexingMode,
+            useTypeName ? null : NpgsqlDbType.TsQuery,
```

Comment:
Not critical for now, but the useTypeName stuff will need to be repeated for every type handler test class. We should ideally just pass NpgsqlDbType.TsQuery and `tsquery`, and the TypeHandlerTestBase should take care of running everything twice - once with the NpgsqlDbType, once with DataTypeName.

Reply:
Don't like it too, probably will rewrite it a bit to handle all cases in a single method, but ideally I would like to separate them into different cases. Unfortunately, nUnit is bad here.